### ISSUE TITLE
[BUG]: Cant get group shuffle split to work under any circumstances

### ISSUE THREAD

**Issue by drewaight** [pycaret/pycaret#3006]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I cannot get group shuffle split to work under any circumstance. I need to use group shuffle split so that rows with the same "label" (group) are not split across train test (otherwise overfitting) I have tried pycaret 2 and 3. 

compare_models runs quickly but does not output results. I have tried defining the groups separately instead of through the dataframe, and also cant even seem to get the inbuilt groupkfold to work either. Is there a way to see where the failure is happening (or output the scikitlearn error?)

![pycaret1](https://user-images.githubusercontent.com/53615928/193294889-b4dac146-1c91-43ca-97fe-00ab6f80c2bf.jpg)
![pycaret2](https://user-images.githubusercontent.com/53615928/193295102-74fe87c9-8aa9-4d8e-94ad-35e9d7e786ad.jpg)
![pycaret3](https://user-images.githubusercontent.com/53615928/193295299-6af665ee-3dd8-4c84-8783-b8fac9d6f79a.jpg)

### Reproducible Example

```python
from pycaret import regression *
from sklearn.model_selection import GroupShuffleSplit
gss = GroupShuffleSplit(n_splits=10, test_size=0.2)

model = setup(data=X, target=targ, train_size=0.8, fold_strategy=gss, fold_groups='labels', silent=True)

models = compare_models()
```

### Expected Behavior

pycaret runs spllits by group

### Actual Results

```python-traceback
compare_models blank output
```

### Installed Versions

2.3.10
3.0.0rc3
3.0.0rc4

**Comment by ninjit** [comment]
I've had the same experience with other CV data splitters.
e.g. the LeaveOneGroupOut (logo) class from sklearn.model_selection

**Comment by ninjit** [comment]
Can you try
` create_model('dummy')`
instead of compare_models()

In my case this revealed the actual error being encountered:
![Screen Shot 2022-10-08 at 19 15 54](https://user-images.githubusercontent.com/753263/194734517-8fee46cc-a9c1-4b22-9dbf-1ff8a740e403.png)

n_splits doesn't seem to be a required attribute for all sklearn CV classes
I think the correct way to query this is with the function `get_n_splits()`, which does seems to be part of all the classes.

**Comment by tvdboom** [comment]
This issue has been fixed in #3057. Feel free to reopen if the issue persists.
